### PR TITLE
fix Loop_out not seeking back

### PR DESCRIPTION
**Pull request by ronso0**
This fixes seeking in the following situation I noticed in https://github.com/mixxxdj/mixxx/pull/12522#issuecomment-1879865194
* enable quantize
* move playhead slightly after a beat
* set `loop_in`
= snaps to previous beat
* jump forward by 4 beats
* set `loop_out`
= snaps to previous beat, playhead moves 4 beats back inside the loop
* deactivate loop
* jump forward by 4 beats
* set `loop_out`
(= previously: snaps to prev. beat, playhead doesn't move, loop will never be reached)
= **FIX:** snaps to previous beat, playhead moves 4 beats back inside the loop

### Note:
As before, (re)setting loop_out with playpos exactly at loop_out of the inactive loop, playpos should stay at loop_out and **not** jump to loop_in.

See discussion on Zulip https://mixxx.zulipchat.com/#narrow/stream/109171-development/topic/2.2E4.20behaviour.20.2F.20failing.20test.20of.20loop_in.2F_out.20with.20quantize

## TODO
- [x] Adjust `LoopInOutButtons_QuantizeEnabled`

### CODE REVIEW

**Comment by ronso0** [comment]
The test `LoopInOutButtons_QuantizeEnabled` should fail now.

**Review by ywwg — APPROVED** [review]
lgtm

**Comment by ywwg** [comment]
extremely annoyingly, the first time I hand-tested it appeared the fix did not work, but every time I tried afterward it did.  Is there a chance that old_loop_info could be blank, but an out point is set?

**Comment by ywwg** [comment]
Unless someone else can reproduce the failure, I think this is a Good Enough fix for the release.  If we find a further edge case later we can fix it in 2.4.1

**Comment by ronso0** [comment]
Draft until the test is fixed.

**Comment by ronso0** [comment]
I disabled the test and filed #12741

File: src/test/looping_control_test.cpp [hunk 1]
```diff
@@ -322,31 +322,47 @@ TEST_F(LoopingControlTest, LoopOutButton_AdjustLoopOutPointInsideLoop) {
     EXPECT_FRAMEPOS_EQ_CONTROL(mixxx::audio::FramePos{1500}, m_pLoopEndPoint);
 }
 
-TEST_F(LoopingControlTest, LoopInOutButtons_QuantizeEnabled) {
+TEST_F(LoopingControlTest, DISABLED_LoopInOutButtons_QuantizeEnabled) {
```
Comment (ywwg) [inline src/test/looping_control_test.cpp:325]: can you add a short comment that links to the issue?

Comment (ronso0) [reply src/test/looping_control_test.cpp:326]: Yup, done.

**Review by ywwg — APPROVED** [review]
